.github/workflows/precommit-crit-flows.yml

```diff
@@ -124,25 +124,75 @@ jobs:
             echo "❌ Deployment failed"; exit 1
           fi
 
+  build_testservice:
+    name: Build Kalium Test Service
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v6
+        with:
+          repository: wireapp/kalium
+          ref: main
+
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Gradle Cache
+        uses: gradle/actions/setup-gradle@4d9f0ba0025fe599b4ebab900eb7f3a1d93ef4c2 # SHA of tag v5.0.0
```

Comment:
Fyi: the only reason I pinned the actions version to a specific commit is because sonar was worried of a supply chain attack. Not sure what's better, having to manually update every time to get e.g. security patches or the risk of malicious code being merged into the action unnoticed.